src/Moq/Setup.cs

```diff
@@ -31,10 +34,12 @@ protected Setup(InvocationShape expectation)
 
 		public bool IsDisabled => (this.flags & Flags.Disabled) != 0;
 
-		public virtual bool IsVerifiable => false;
+		public bool IsVerifiable => (this.flags & Flags.Verifiable) != 0;
 
 		public MethodInfo Method => this.expectation.Method;
 
+		public Mock Mock => this.mock;
```

Comment:
Could be exposed via `ISetup`. The only reason why I opted not to do so right away is because we don't link back to the owning `Mock` in `IInvocation`, either (in order to keep that type and its implementation slim)... nor do we have back links to the `Mock` in the `mock.Invocations` and `mock.Setups` collections. It's nice to have some consistency across all of these types.